python/integrationtest/log_file_checks.py

```diff
@@ -1,31 +1,74 @@
     ok=True
     for line in open(log_file_name).readlines():
 
         # First check if the line appears to be in the standard format of messages produced with our logging package
         # For lines produced with our logging package, the first two words in the line are the date and time, then the severity
 
+        bad_line=False
         match_logline_prefix = re.search(r"^20[0-9][0-9]-[A-Z][a-z][a-z]-[0-9]+\s+[0-9:,]+\s+([A-Z]+)", line)
         if match_logline_prefix:
             severity=match_logline_prefix.group(1)
             if severity in ("WARN", "ERROR", "FATAL"):
-                print(line)
-                ok=False
+                bad_line=True
         else: # This line's not produced with our logging package, so let's just look for bad words
             if "WARN" in line or "Warn" in line or "warn" in line or \
                "ERROR" in line or "Error" in line or "error" in line or \
                "FATAL" in line or "Fatal" in line or "fatal" in line or \
                "egmentation fault" in line:
-                print(line)
-                ok=False
-            
+                bad_line=True
+
+        if bad_line:
+            ignore_this_problem=False
+            for excluded_substring in excluded_substring_list:
+                if excluded_substring in line:
+                    ignore_this_problem=True
+                    break
+            if ignore_this_problem:
+                bad_line=False
+        if bad_line:
+            if ok and print_logfilename_for_problems:
+                print("----------")
+                print(f"Problem(s) found in logfile {log_file_name}:")
+            print(line)
+            ok=False
     return ok
 
-def logs_are_error_free(log_file_names):
+# 14-Nov-2021, KAB: added the ability for users to specify sets of excluded substrings, to
+# enable checking of all log files, and to print out the logfile name when there are problems.
+#
+# This function accepts the following arguments:
+# * the list of logfiles to be checked (array of PythonPath objects)
+# * a flag to control whether all logfiles are checked for problems or checking stops as
+#   soon as one file with problems is found (default is to stop after the first one)
+# * a flag to control whether the logfile name is printed to the console when an a problem
+#   is first found in that logfile (default is no printout)
+# * the sets of excluded substrings.  This goal of this argument is to allow certain
+#   select messages to be ignored so that overall checking of logfiles can remain enabled
+#   without being distracted by 'expected' problems.  This argument is expected to be a
+#   dictionary keyed by strings that might appear in the logfile name and having values
+#   that are lists of excluded phrases. For example:
+#   ex_sub_map = {"ruemu": ["expected problem phrase 1", "expected problem  phrase 2"]}
+def logs_are_error_free(log_file_names, show_all_problems=False, print_logfilename_for_problems=False,
```

Comment:
Do you want to make `show_all_problems` and `print_logfilename_for_problems` both default to `True`? I think that would be reasonable